src/librustc_driver/lib.rs

```diff
@@ -126,12 +128,12 @@ pub fn run_compiler(
     callbacks: &mut (dyn Callbacks + Send),
     file_loader: Option<Box<dyn FileLoader + Send + Sync>>,
     emitter: Option<Box<dyn Write + Send>>
-) -> interface::Result<()> {
+) -> interface::Result<bool> {
```

Comment:
Instead of changing the return type here, use `Callbacks` to find out if timings are enabled.

Reply:
I don't understand this suggestion. We pass in a `DefaultCallbacks` instance, and that type implements the `Callback` trait via the default ops, which are no-ops. What would the code look like?

Reply:
Just create a new struct (with a bool field) and implement the `Callbacks` trait for it. Override the `config` method to read out if timings are enabled.

Reply:
Returning the bool is a lot simpler. No traits involved!

Reply:
It also doesn't make sense for `run_compiler` to return just a random `bool` though, and this also breaks RLS and Miri for  no reason.